### PR TITLE
Restructuring the DigitalObject override to use the Pulfalight namespace

### LINKED ISSUE #245: Restructure how app/overrides is used to override the DigitalObject Class
Also, we are currently using the `ArcLight` namespace, should we be doing that?

### PR DESCRIPTION
**Pull request by jrgriffiniii**
Resolves #245

### CODE REVIEW

File: lib/pulfalight/digital_object.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 # Add a role attribute to this class
-class Arclight::DigitalObject
+class Pulfalight::DigitalObject < Arclight::DigitalObject
```
Comment (tpendragon) [inline lib/pulfalight/digital_object.rb:4]: Why move this to lib, out of curiosity? Now we have models in two places. Does this work in like, `app/models/pulfalight/digital_object.rb`?

File: config/application.rb [hunk 2]
```diff
@@ -18,12 +18,7 @@ class Application < Rails::Application
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
 
-    # load overrides
-    config.to_prepare do
```
Comment (tpendragon) [inline config/application.rb:22]: Isn't there another override?

**Comment by hackmastera** [comment]
@jrgriffiniii is it possible that spring was on and causing problems? Could the fix be as simple as turning off spring?

File: lib/pulfalight/digital_object.rb (hunk 1 above)
Comment (jrgriffiniii) [reply lib/pulfalight/digital_object.rb:4]: I was following the pattern in https://github.com/projectblacklight/arclight/blob/master/lib/arclight/digital_object.rb

File: config/application.rb (hunk 2 above)
Comment (jrgriffiniii) [reply config/application.rb:22]: I am sorry, I can adjust this.

**Comment by jrgriffiniii** [comment]
@hackmastera I will try using a fresh clone and restarting spring.

**Comment by jrgriffiniii** [comment]
@hackmastera I apologize, but stopping spring entirely still raised the error. The document against which I'm testing is `mudd/publicpolicy/MC104.EAD.xml`, and this error remains:

```
ArgumentError: unknown keyword: role
bundler/gems/arclight-6020c4ef4f65/lib/arclight/digital_object.rb:8:in `initialize'
```

I could try keeping this in place and just fixing the line on https://github.com/pulibrary/pulfalight/blob/master/lib/pulfalight/traject/ead2_config.rb#L13, but I just wanted to be certain that it might not be an issue elsewhere.

File: config/application.rb (hunk 2 above)
Comment (hackmastera) [reply config/application.rb:22]: There weren't any others.

**Comment by hackmastera** [comment]
Oh, I see. The override load wasn't having an affect on the lib directory.

I think it would be safe to just fix it in that file, since anything in lib will always need an explicit require anyway.

**Comment by jrgriffiniii** [comment]
Deprecated by #250

**Pull request closed without merging**